protostar/self/cache_io_test.py

```diff
@@ -21,6 +21,9 @@ def test_cache_simple(tmp_path):
 
     assert cache_io.read(cache_name) is None
 
+    # pylint: disable=protected-access
+    assert cache_io._gitignore_path.read_text(encoding="utf-8") == "*\n"
```

Comment:
Either:
1. Consider making this field public
2. Preferred: assume the path in the test (use hardcoded value instead of reading the field)

Reply:
Why exactly is this code a problem?
1. I can do that
2. I think this is a no-go, because `_gitignore_path` evaluates to `Path(self._cache_path / ".gitignore")`, which evaluates to `Path((project_root_path / Path(self._CACHE_DIR_NAME)) / ".gitignore")`, so we'd eventually have to make something public anyway (in this case `_CACHE_DIR_NAME`)

I think that accessing private members in tests is okay - you want to test if everything is okay at the same having a clean structure in the prod.

Reply:
> I think that accessing private members in tests is okay - you want to test if everything is okay at the same having a clean structure in the prod.

I don't fully share this opinion, as I believe in writing tests that test a black box which exposes a public API. I think exposing a property pointing to the cache directory would make sense.